crates/router/src/connector/adyen/transformers.rs

```diff
@@ -836,18 +836,31 @@ fn get_address_info(address: Option<&api_models::payments::Address>) -> Option<A
 }
 
 fn get_line_items(item: &types::PaymentsAuthorizeRouterData) -> Vec<LineItem> {
-    let order_details = item.request.order_details.as_ref();
-    let line_item = LineItem {
-        amount_including_tax: Some(item.request.amount),
-        amount_excluding_tax: Some(item.request.amount),
-        description: order_details.map(|details| details.product_name.clone()),
-        // We support only one product details in payment request as of now, therefore hard coded the id.
-        // If we begin to support multiple product details in future then this logic should be made to create ID dynamically
-        id: Some(String::from("Items #1")),
-        tax_amount: None,
-        quantity: Some(order_details.map_or(1, |details| details.quantity)),
-    };
-    vec![line_item]
+    let order_details = item.request.order_details.clone();
+    match order_details {
+        Some(od) => od
+            .iter()
+            .map(|data| LineItem {
+                amount_including_tax: Some(item.request.amount),
+                amount_excluding_tax: Some(item.request.amount),
+                description: Some(data.product_name.clone()),
+                id: Some(String::from("Items #1")),
+                tax_amount: None,
+                quantity: Some(data.quantity),
+            })
+            .collect(),
+        None => {
+            let line_item = LineItem {
+                amount_including_tax: Some(item.request.amount),
+                amount_excluding_tax: Some(item.request.amount),
+                description: None,
+                id: Some(String::from("Items #1")),
+                tax_amount: None,
+                quantity: None,
```

Comment:
This should be 1 by default i guess by looking at the previous code.

Reply:
changed to :
`quantity: Some(1),`